pkg/middlewares/headers/headers.go

```diff
@@ -178,6 +187,24 @@ func (s *Header) modifyRequestHeaders(req *http.Request) {
 	}
 }
 
+func (s *Header) inlineModifyResponseHeaders(rw http.ResponseWriter, req *http.Request) {
```

Comment:
maybe we can add something like:

// inlineModifyResponseHeaders sets during request processing time, all the response headers that we already know that are bound to be set, and which do not depend on a later state of the response. One notable example of a header that can only be modified later on is "Vary".

Or we could do it above L164